components/drawer/index.tsx

```diff
@@ -61,271 +66,270 @@ interface InternalDrawerProps extends DrawerProps {
 }
 
 const defaultPushState: PushState = { distance: 180 };
-class Drawer extends React.Component<InternalDrawerProps, IDrawerState> {
-  static defaultProps = {
-    width: 256,
-    height: 256,
-    closable: true,
-    placement: 'right' as placementType,
-    maskClosable: true,
-    mask: true,
-    level: null,
-    keyboard: true,
-    push: defaultPushState,
-  };
-
-  readonly state = {
-    push: false,
-  };
-
-  parentDrawer: Drawer | null;
-
-  destroyClose: boolean;
-
-  public componentDidMount() {
-    // fix: delete drawer in child and re-render, no push started.
-    // <Drawer>{show && <Drawer />}</Drawer>
-    const { visible } = this.props;
-    if (visible && this.parentDrawer) {
-      this.parentDrawer.push();
-    }
-  }
-
-  public componentDidUpdate(preProps: DrawerProps) {
-    const { visible } = this.props;
-    if (preProps.visible !== visible && this.parentDrawer) {
-      if (visible) {
-        this.parentDrawer.push();
-      } else {
-        this.parentDrawer.pull();
+const defaultCloseIcon = <CloseOutlined />;
+// React.Component<InternalDrawerProps, IDrawerState>
```

Comment:
removed

Reply:
I don't understand, Could you explain what can i do?